src/catalog/src/information_schema/region_peers.rs

```diff
@@ -0,0 +1,275 @@
+            peer_ids: UInt64VectorBuilder::with_capacity(42),
+            peer_addrs: StringVectorBuilder::with_capacity(42),
+            is_leaders: StringVectorBuilder::with_capacity(42),
+            statuses: StringVectorBuilder::with_capacity(42),
+            down_seconds: Int64VectorBuilder::with_capacity(42),
+        }
+    }
+
+    /// Construct the `information_schema.region_peers` virtual table
+    async fn make_region_peers(&mut self, request: Option<ScanRequest>) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
+        let catalog_manager = self
+            .catalog_manager
+            .upgrade()
+            .context(UpgradeWeakCatalogManagerRefSnafu)?;
+
+        let partition_manager = catalog_manager
+            .as_any()
+            .downcast_ref::<KvBackendCatalogManager>()
+            .map(|catalog_manager| catalog_manager.partition_manager());
+
+        let predicates = Predicates::from_scan_request(&request);
+
+        for schema_name in catalog_manager.schema_names(&catalog_name).await? {
+            if !catalog_manager
+                .schema_exists(&catalog_name, &schema_name)
+                .await?
+            {
+                continue;
+            }
+
+            let mut stream = catalog_manager.tables(&catalog_name, &schema_name).await;
+
+            while let Some(table) = stream.try_next().await? {
+                let table_info = table.table_info();
+
+                if table_info.table_type == TableType::Temporary {
+                    continue;
+                }
+
+                let table_id = table_info.ident.table_id;
+                let routes = if let Some(partition_manager) = &partition_manager {
+                    partition_manager
+                        .find_region_routes(table_id)
+                        .await
+                        .context(FindRegionRoutesSnafu {
+                            table: &table_info.name,
+                        })?
+                } else {
+                    // Standalone doesn't has route values.
+                    vec![]
+                };
+
+                self.add_region_peers(&predicates, &routes);
+            }
+        }
+
+        self.finish()
+    }
+
+    fn add_region_peers(&mut self, predicates: &Predicates, routes: &[RegionRoute]) {
+        for route in routes {
+            let region_id = route.region.id.as_u64();
+            let peer_id = route.leader_peer.clone().map(|p| p.id);
+            let peer_addr = route.leader_peer.clone().map(|p| p.addr);
+            let status = if let Some(status) = route.leader_status {
+                Some(status.name().to_string())
+            } else {
+                // Alive by default
+                Some("ALIVE".to_string())
+            };
+
+            let row = [(REGION_ID, &Value::from(region_id))];
+
+            if !predicates.eval(&row) {
+                return;
+            }
```

Comment:
What criteria do we use to decide which columns can be used for filtering?

Reply:
Just a personal judgment that I think is the most common query criteria we may use.